src/source/image_source.js

```diff
@@ -207,7 +201,7 @@ class ImageSource extends Evented implements Source {
             this.texture.bind(gl.LINEAR, gl.CLAMP_TO_EDGE);
         } else if (resize) {
             this.texture.update(image);
-        } else if (image instanceof window.HTMLVideoElement || image instanceof window.ImageData || image instanceof window.HTMLCanvasElement) {
+        } else if (image instanceof window.HTMLVideoElement || image instanceof window.HTMLCanvasElement) {
```

Comment:
per your comment, should we also check if `animate: true` for `CanvasSource` in this conditional?

Reply:
Good call, added a fix for that as well.